pkg/gameserverallocations/controller.go

```diff
@@ -398,26 +398,32 @@ func (c *Controller) allocateFromRemoteCluster(gsa v1alpha1.GameServerAllocation
 	}
 
 	// TODO: Retry on transient error --> response.StatusCode >= 500
-	response, err := client.Post(connectionInfo.AllocationEndpoint, "application/json", bytes.NewBuffer(body))
-	if err != nil {
-		return nil, err
-	}
-	defer response.Body.Close() // nolint: errcheck
+	for i, endpoint := range connectionInfo.AllocationEndpoints {
+		response, err := client.Post(endpoint, "application/json", bytes.NewBuffer(body))
+		if err != nil {
+			return nil, err
+		}
+		defer response.Body.Close() // nolint: errcheck
 
-	data, err := ioutil.ReadAll(response.Body)
-	if err != nil {
-		return nil, err
-	}
-	if response.StatusCode >= 400 {
-		// For error responses return the body without deserializing to an object.
-		return nil, errors.New(string(data))
-	}
+		data, err := ioutil.ReadAll(response.Body)
+		if err != nil {
+			return nil, err
+		}
+		if response.StatusCode >= 500 && (i+1) < len(connectionInfo.AllocationEndpoints) {
```

Comment:
```suggestion
		if response.StatusCode >= 500 {
```
I think that this check is redundant for loop will check not continue if we are out of `range connectionInfo.AllocationEndpoints`.

Reply:
Thanks. This condition says if the there is any other endpoint try sending the request with the new endpoint. Otherwise return the error. If I remove the condition, no error will be returned